doc/mgmt/SONiC_OC_PortChannel_Interface_HLD.md

```diff
@@ -0,0 +1,439 @@
++         |  |  +--ro oc-lag:aggregate-id?             -> /oc-if:interfaces/interface/name
++         +--rw oc-lag:aggregation
++         |  +--rw oc-lag:config
+          |  |  +--rw oc-lag:lag-type?    aggregation-type
++         |  |  +--rw oc-lag:min-links?   uint16
++         |  +--ro oc-lag:state
+          |  |  +--ro oc-lag:lag-type?    aggregation-type
++         |  |  +--ro oc-lag:min-links?   uint16
+          |  |  +--ro oc-lag:lag-speed?   uint32
++         |  |  +--ro oc-lag:member*      oc-if:base-interface-ref
++         |  |  +--ro if-agmnt:fallback?  boolean  //Augmented
+
+```
+
+**Note:** openconfig-if-aggregate.yang data model is augmented to support LACP Fallback.
+
+### 3.6.2 CLI
+#### 3.6.2.1 Configuration Commands
+
+### Create a PortChannel
+`interface PortChannel <channel-number>`
+```
+sonic(config)# interface PortChannel 1
+```
+### Configure min-links
+`minimum-links <number>`
+<br>
+ Default:0
+```
+sonic(config)# interface PortChannel 1
+sonic(conf-if-po1)# minimum-links 1
+```
+### Reset to default value of min-links
+`no minimum-links`
+<br>
+ Default:0
+```
+sonic(conf-if-po1)# no minimum-links 
+```
+### Configure MTU
+`mtu <mtu-val>`
+```
+sonic(conf-if-po1)#  mtu 9000
+sonic(conf-if-po1)# no mtu
+```
+#### MTU Removal
+`no mtu` --> Reset to default
+```
+sonic(conf-if-po1)# no mtu
+```
+
+#### Enable PortChannel
+`no shutdown` 
+```
+sonic(conf-if-po1)# no shutdown
+```
+#### Disable PortChannel
+`shutdown` 
+```
+sonic(conf-if-po1)# shutdown
+```
+
+#### Configures an IPv4 address 
+`ip address <ip-address with mask> | no ip address <ip-address>`
+```
+sonic(conf-if-po1)# ip address 2.2.2.2/24
+```
+#### Remove IPv4 address 
+`no ip address <ip-address>`
+```
+sonic(conf-if-po1)# no ip address 2.2.2.2
+```
+#### Configures an IPv6 address of the interface.
+`ipv6 address <ipv6-address with mask> | no ipv6 address <ipv6-address>`
+```
+sonic(config)# interface Ethernet 4
+sonic(conf-if-Ethernet4)# ipv6 address a::e/64
+sonic(conf-if-Ethernet4)# no ipv6 address a::e
+```
+#### Add port member
+`channel-group <channel-number>`
+```
+sonic(config)# interface Ethernet4
+sonic(conf-if-Ethernet4)# channel-group 1
+```
+#### Remove a port member
+`no channel-group`
+```
+sonic(config)# interface Ethernet4
+sonic(conf-if-Ethernet4)# no channel-group
+```
+#### Delete a PortChannel
+`no interface PortChannel <channel-number>`
+```
+sonic(config)# no interface PortChannel 1
+```
+
+#### 3.6.2.2 Show Commands
+
+##### Display summary information about PortChannels
+`show PortChannel summary`
+```
+sonic# show PortChannel summary
+
+Flags:  D - Down
+        U - Up
+--------------------------------------------------------------------------------
+Group Port-Channel           Type     Protocol  Member Ports
+--------------------------------------------------------------------------------
+1    PortChannel1    (D)     Eth      DYNAMIC    Ethernet56(D), Ethernet60(D)
```

Comment:
1. I went through the output of "show port-channel summary" of other vendors in the industry. This output is not 100% inline with them.
2. What does Type mean here?
3. The fields in Protocol column should be - LACP/NONE.
4. The member port status in the parenthesis should be P (if port is selected), empty (if port is de-selected).

Reply:
#5. Currently SONiC does not show short forms for interfaces. 
#4. We can update (P,I) for selected, inactive port. However those details come from teamd. 
#3. This will show Static/Dynamic. The protocol will always be LACP. We could also remove this line as Buzznik will only have dynamic.

Reply:
#2 Since the Type value always remained "Eth" I have removed that column from show PortChannel sumamry output.

Reply:
#5 - I agree. But I guess we're trying to match our CLI with industry standard with the current changes. So we can try to be inline to the maximum possible.
#4 - Though these details come from teamd, the interpretation of these values is done in SONiC (currently done in teamshow file). We can always change them to meet the industry standard CLI.
#3 - Port-channel protocol means which protocol is being used; Static/Dynamic are not protocols, they are types of the port-channel. Broadcom is implementing static LAG in Buzznik; so its better to have a placeholder to dispaly the protocol. 
#2 - Lets try to match the output with other vendors in the industry.

Reply:
#3 I have followed the OS10 way of displaying output, currently it displays DYNAMIC/STATIC under protocol column, I can change it to LACP/NONE for Buzznik release.